airflow/api/client/api_client.py

```diff
@@ -31,14 +36,23 @@ def __init__(self, api_base_url, auth=None, session: httpx.Client | None = None)
         if auth:
             self._session.auth = auth
 
-    def trigger_dag(self, dag_id, run_id=None, conf=None, execution_date=None, replace_microseconds=True):
+    def trigger_dag(
+        self,
+        dag_id,
+        run_id=None,
+        conf=None,
+        execution_date=None,
+        replace_microseconds=True,
+        triggered_by: DagRunTriggeredByType | None = None,
```

Comment:
What does a `None` value mean? I feel we should make this argument required instead.

Reply:
Actually, my initial thought was setting None here as default but setting the default value for this property has a different value like `REST_API` in the child classes. But it also makes sense to convert this parameter to a required one (I even forget to set the default values in child classes)

Also, is it OK to have this change or I need to follow @Taragolis's recommendation and revert the changes in this folder: `airflow/api/client/*`; WDYT?

Reply:
Let’s revert the changes in `api/client` first.